gufe/protocols/protocolunit.py

```diff
@@ -30,7 +32,7 @@ class Context:
 
     """
     scratch: PathLike
-    shared: PathLike
+    shared: ExternalStorage
```

Comment:
may as well make both `ExternalStorage` at this point

Reply:
Disagree. I think `scratch` is inherently a local directory. There is never a need for it to be anything else (a protocol unit must be run on a single node). Also, I think that requiring contributors to use `get_filename` for every file in their protocol is going to be tedious, especially when they are porting existing code to be gufe-compatible.

Current model is that they use `get_filename` to obtain files that are inputs to the protocol. We might be able to simplify that, though, so they don't even need that. Writing output is done because the outputs are identified from `pathlib.Path` objects in the return dict from `_execute`.